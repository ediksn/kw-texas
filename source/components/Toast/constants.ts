import { verticalScale } from 'react-native-size-matters'

export const TOAST_HEIGHT = verticalScale(48)
export const TOAST_TYPE = {
  ERROR: 'ERROR',
  SUCCESS: 'SUCCESS'
}
