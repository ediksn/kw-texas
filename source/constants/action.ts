export const ACTION = {
  APP: {
    LOGIN: 'ACTION.APP.LOGIN',
    LOGOUT: 'ACTION.APP.LOGOUT'
  }
}
