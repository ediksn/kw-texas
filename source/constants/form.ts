export const FORM: any = {
  FIELDS_TYPES: {
    TEXT: 'TEXT',
    PASSWORD: 'PASSWORD'
  }
}
