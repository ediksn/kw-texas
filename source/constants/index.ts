export { FORM } from './form'
export { NAVIGATION } from './navigation'
export { theme } from './theme'
export { ACTION } from './action'
export * from './statis'
export * from './testIds'
