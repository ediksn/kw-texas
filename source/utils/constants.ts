export const MAX_CHARACTERS_NEW_POST: number = 500
export const POST_PREVIEW_MEDIA_LENGTH: number = 3
export const GROUP_PREVIEW_MEDIA_LENGTH: number = 10
