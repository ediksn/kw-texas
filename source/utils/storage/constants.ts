export default {
  LOGIN: {
    USER: 'LOGIN.USER',
    SESSION: 'LOGIN.SESSION'
  },
  BIOMETRIC: {
    PERMISSION: 'BIOMETRIC.PERMISSION',
    CREDENTIALS: 'BIOMETRIC.CREDENTIALS'
  }
}
