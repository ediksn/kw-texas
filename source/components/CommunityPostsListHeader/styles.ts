import { StyleSheet } from 'react-native'
import { moderateScale, verticalScale } from 'react-native-size-matters'
import { theme } from '~/constants'

export const styles = StyleSheet.create({
  typeSomething: {
    fontSize: theme.fonts.MEDIUM_SIZE,
    color: theme.darkGreenColor,
    marginLeft: moderateScale(10),
    fontFamily: 'Mulish-Bold'
  },
  headerListContainer: {
    backgroundColor: theme.greenLightColor,
    marginBottom: verticalScale(5),
    paddingHorizontal: moderateScale(18)
  },
  headerCard: {
    elevation: 4,
    backgroundColor: '#FFFFFF',
    borderRadius: 4,
    marginBottom: moderateScale(5),
    paddingVertical: verticalScale(5),
    marginTop: moderateScale(5),
    marginHorizontal: moderateScale(2),
    shadowColor: theme.post.shadowColor,
    shadowOffset: {
      width: 0,
      height: 3
    },
    shadowOpacity: 0.23,
    shadowRadius: 3
  },
  typeSomethingRow: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingHorizontal: moderateScale(10),
    flex: 1
  }
})
