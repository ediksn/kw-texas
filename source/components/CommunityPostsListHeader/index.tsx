import { useTranslation } from 'react-i18next'
import { Text, View, TouchableOpacity } from 'react-native'
import React from 'react'
import { useNavigation } from '@react-navigation/native'
import { styles } from './styles'
import Avatar from '~/components/Avatar'
import { NAVIGATION } from '~/constants/navigation'
import { OptionInterface } from '~/interfaces/groupInterface'
import { TEST_IDS } from '~/constants'

interface ListHeaderComponentProps {
  avatarUri: string
  currentGroupSelected: OptionInterface
}

const CommunityPostsListHeader = ({ avatarUri, currentGroupSelected }: ListHeaderComponentProps) => {
  const { t } = useTranslation()
  const navigation = useNavigation()

  return (
    <View testID={TEST_IDS.COMMUNITY.HOME.HEADER} style={styles.headerListContainer}>
      <View style={styles.headerCard}>
        <View style={styles.typeSomethingRow}>
          <Avatar uri={avatarUri} />
          <TouchableOpacity
            onPress={() =>
              navigation.navigate(NAVIGATION.SCREEN.NEWPOST, { fromCommunity: true, currentGroupSelected })
            }
          >
            <Text style={styles.typeSomething}>{t('components_NewPost_Share_today')}</Text>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  )
}

export default CommunityPostsListHeader
