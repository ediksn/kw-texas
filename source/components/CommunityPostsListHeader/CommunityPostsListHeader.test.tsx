import React from 'react'
import { Provider } from 'react-redux'
import { cleanup, RenderAPI, render } from '@testing-library/react-native'
import createTestStore from '../../../__mocks__/store'
import CommunityPostsListHeader from './index'
import { TEST_IDS } from '~/constants'

afterEach(cleanup)

jest.mock('@react-navigation/native', () => ({
  createNavigatorFactory: jest.fn(),
  useNavigation: () => ({
    navigate: jest.fn(),
    dispatch: jest.fn(),
    goBack: jest.fn(),
    setOptions: jest.fn()
  }),
  useRoute: () => ({
    params: {
      editMode: false,
      idPost: '1',
      groupId: '1'
    }
  }),
  useFocusEffect: jest.fn()
}))

jest.mock('@react-navigation/material-top-tabs', () => ({
  createMaterialTopTabNavigator: () => ({
    Navigator: jest.mock,
    Screen: jest.mock
  })
}))

jest.mock('@react-navigation/stack', () => ({
  createStackNavigator: () => ({
    Navigator: jest.mock,
    Screen: jest.mock
  })
}))

jest.mock('@react-navigation/bottom-tabs', () => ({
  createBottomTabNavigator: () => ({
    Navigator: jest.mock,
    Screen: jest.mock
  })
}))

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: any) => key })
}))

describe('Render Community Detail Test', () => {
  let component: RenderAPI
  let storeReducers
  const currentGroupSelectedMock = {
    key: '0001',
    isTitle: true,
    handleOption: () => null,
    title: 'Title Mock',
    color: '#ffffff'
  }

  beforeEach(() => {
    storeReducers = createTestStore()
    component = render(
      <Provider store={storeReducers}>
        <CommunityPostsListHeader avatarUri='' currentGroupSelected={currentGroupSelectedMock} />
      </Provider>
    )
  })

  it('Match with the snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('Renders correctly', () => {
    expect(component).toBeDefined()
    expect(component.queryAllByTestId(TEST_IDS.COMMUNITY.HOME.HEADER).length).toEqual(1)
  })
})
