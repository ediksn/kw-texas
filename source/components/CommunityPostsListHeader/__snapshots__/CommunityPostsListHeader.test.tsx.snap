// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Community Detail Test Match with the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F7F7F8",
      "marginBottom": 9.808823529411764,
      "paddingHorizontal": 28.285714285714285,
    }
  }
  testID="communityHeader"
>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 4,
        "elevation": 4,
        "marginBottom": 7.857142857142857,
        "marginHorizontal": 3.142857142857143,
        "marginTop": 7.857142857142857,
        "paddingVertical": 9.808823529411764,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.23,
        "shadowRadius": 3,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "paddingHorizontal": 15.714285714285714,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Array [
            Object {
              "height": 62.857142857142854,
              "opacity": 0.5,
              "width": 62.857142857142854,
            },
            undefined,
          ]
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#006C84",
              "fontFamily": "Mulish-Bold",
              "fontSize": 22,
              "marginLeft": 15.714285714285714,
            }
          }
        >
          components_NewPost_Share_today
        </Text>
      </View>
    </View>
  </View>
</View>
`;
